fuel-p2p/src/discovery.rs

```diff
@@ -0,0 +1,507 @@
+use self::mdns::MdnsWrapper;
+use futures::FutureExt;
+use futures_timer::Delay;
+use ip_network::IpNetwork;
+use libp2p::{
+    core::{
+        connection::{ConnectionId, ListenerId},
+        ConnectedPoint,
+    },
+    kad::{handler::KademliaHandlerProto, store::MemoryStore, Kademlia, KademliaEvent, QueryId},
+    mdns::MdnsEvent,
+    multiaddr::Protocol,
+    swarm::{
+        DialError, IntoProtocolsHandler, NetworkBehaviour, NetworkBehaviourAction, PollParameters,
+        ProtocolsHandler,
+    },
+    Multiaddr, PeerId,
+};
+use log::trace;
+use std::{
+    collections::{HashMap, HashSet, VecDeque},
+    io,
+    task::{Context, Poll},
+    time::Duration,
+};
+mod discovery_config;
+mod mdns;
+pub use discovery_config::DiscoveryConfig;
+
+const SIXTY_SECONDS: Duration = Duration::from_secs(60);
+
+/// Event generated by the `DiscoveryBehaviour`.
+#[derive(Debug)]
+pub enum DiscoveryEvent {
+    /// Notify the swarm of an UnroutablePeer
+    UnroutablePeer(PeerId),
+
+    /// Notify the swarm of a connected peer
+    Connected(PeerId),
+
+    /// Notify the swarm of a disconnected peer
+    Disconnected(PeerId),
+}
+
+/// NetworkBehavior for discovery of nodes
+pub struct DiscoveryBehaviour {
+    /// Store Peer IDs of currently connected peers
```

Comment:
now that I'm working on PeerInfo Behaviour, I might move `connected_peers` and `peer_addresses` there

Reply:
Would it be easier to move them over after this PR gets merged?

Reply:
yes, that's the idea - I just left a comment here as a _heads up_